Add META.* and Build.PL

zakame opened the pull request:
This PR is a submission for my [CPAN Pull Request Challenge this month](http://cpan-prc.org/2018/september.html).

Add stuff generated by [mbtiny](https://metacpan.org/pod/distribution/App-ModuleBuildTiny/script/mbtiny) to allow installing with `cpanm` and friends by referring to the Git repository directly, e.g.

    $ cd Hijk $ cpanm .

or even

    $ cpanm git://github.com/gugod/Hijk.git

zakame committed:
Add META.* and Build.PL

Add stuff generated by mbtiny to allow installing with `cpanm` and
friends by referring to the Git repository directly, e.g.

    $ cd Hijk
    $ cpanm .

or even

    $ cpanm git://github.com/gugod/Hijk.git

gugod reviewed (changes requested):
Thanks for the initiative. Looks mostly good to me ! If you could fix the issues messy characters, I'll merged this PR 👍

zakame commented on META.json:
```diff
@@ -0,0 +1,64 @@
+   "author" : [
+      "=over 4",
+      "=item Kang-min Liu <gugod@gugod.org>",
+      "=item ÃÂvar ArnfjÃÂ¶rÃÂ° Bjarmason <avar@cpan.org>",
```
Thanks for the review!

Looks like this is due to https://github.com/gugod/Hijk/blob/2804423ed3e8f36d93168985cbec319b002a8eaf/lib/Hijk.pm#L718-L731

Which in turn App::ModuleBuildTiny::Dist tries to parse in https://github.com/Leont/app-modulebuildtiny/blob/74275cf4f9b7260b0c5496dc9ac6c098f6748dc9/lib/App/ModuleBuildTiny/Dist.pm#L130-L131 (through https://metacpan.org/pod/Module::Metadata.)

There are two problems here:

- extra pod is included
- App::ModuleBuildTiny::Dist doesn't seem to emit files in Unicode (considering `Hijk.pm` is already in Unicode and the pod is explicitly set `=encoding utf8`)

For the first problem, an easy fix might be just to reformat the `AUTHORS` pod to something like this:

```perl
=head1 AUTHORS

Kang-min Liu <gugod@gugod.org>

Ævar Arnfjörð Bjarmason <avar@cpan.org>

Borislav Nikolov <jack@sofialondonmoskva.com>

Damian Gryski <damian@gryski.com>

=cut
```

The second problem seems an issue in App::ModuleBuildTiny::Dist itself, so I think I can raise an issue there and link to this.  Meanwhile, I can change this file here directly to unmunge Ævar's name.

zakame committed:
Hijk.pm: Fix AUTHORS pod

Reformat so Module::Metadata (and authoring tools on top) can read it in
without extra pod.

zakame committed:
Regenerate and fix META.* files

It seems that `mbtiny generate` writes without setting the right
encoding, leading to Ævar's name getting munged.

gugod reviewed (approved):
Thanks ! 👍

zakame commented:
Thanks too! :tada:
